guides/source/active_storage_overview.md

```diff
@@ -539,60 +539,76 @@ Analysis requires the `mini_magick` gem. Video analysis also requires the [FFmpe
 
 [`analyzed?`]: https://api.rubyonrails.org/classes/ActiveStorage/Blob/Analyzable.html#method-i-analyzed-3F
 
-Transforming Images
--------------------
+Displaying Images, Videos, and PDFs
 
-To enable variants, add the `image_processing` gem to your `Gemfile`:
+Active Storage supports [representing][] a variety of files. You can use the `representation` method on an attachment to display an image variant, or a preview of a video or PDF. Before calling `representation`, check if the attachment can be represented by calling `representable?`. Some file formats can't be previewed natively in the browser (eg. Word documents); if `representable?` returns false you may want to [link to](#linking-to-files) the file instead.
+
+```erb
+<ul>
+  <% @message.files.each do |file| %>
+    <li>
+      <% if file.representable? %>
+        <%= image_tag file.representation(resize_to_limit: [100, 100]) %>
+      <% else %>
+        <%= link_to url_for(file) do %>
+          Download file
+        <% end %>
+      <% end %>
+    </li>
+  <% end %>
+</ul>
+```
+
+Internally, `representation` calls `variant` for images, and `preview` for previewable files. You can also call these methods directly.
+
+[representing]: https://api.rubyonrails.org/classes/ActiveStorage/Blob/Representable.html
+
+### Transforming Images
+
+Transforming images allows you to display the image at dimensions of your choosing. To enable variants, add the `image_processing` gem to your `Gemfile`:
 
 ```ruby
 gem 'image_processing'
 ```
 
-To create a variation of an image, call [`variant`][] on the attachment. You can pass any transformation to the method supported by the processor. The default processor for Active Storage is MiniMagick, but you can also use [Vips](https://www.rubydoc.info/gems/ruby-vips/Vips/Image).
-
-When the browser hits the variant URL, Active Storage will lazily transform the
-original blob into the specified format and redirect to its new service
-location.
+To create a variation of an image, call [`variant`][] on the attachment. You can pass any transformation to the method supported by the processor. When the browser hits the variant URL, Active Storage will lazily transform the original blob into the specified format and redirect to its new service location.
 
 ```erb
 <%= image_tag user.avatar.variant(resize_to_limit: [100, 100]) %>
 ```
 
-To switch to the Vips processor, you would add the following to
-`config/application.rb`:
+If you don't want to transform images and just want the file that was uploaded, [link to the file](#linking-to-files).
+
+The default processor for Active Storage is MiniMagick, but you can also use [Vips][]. To switch to Vips, add the following to `config/application.rb`:
```

Comment:
I think it's fine to put the Vips URL inline here.  In PRs like #40706, I use reference-style links because: (1) the URLs are very long; (2) I don't want to rewrap (or otherwise break the wrapping of) existing text; and (3) API methods are occasionally linked in multiple places, when it suits the surrounding text.  But none of these really apply to the Vips link.

Reply:
True, but on the other hand, having every link be done this way makes it easier for docs writers to reason about. I think it's better to not have to memorise that checklist when writing docs, and the easiest way to do that is to treat each link the same.